### PR TITLE
feat: [PROD-12043] create dataset with AzureStorage

### PR DESCRIPTION
**Pull request by esasova**
Before creating a PR, please check that:
- [ ] Code is clean
- [ ] Tests are still working (cypress tests and `yarn test`)
- [ ] Changes don't cause new errors or warnings in browser console
- [ ] Documentation is up-to-date
- [ ] Breaking changes are clearly identified with [conventional commits](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index)

### CODE REVIEW

File: public/locales/fr/translation.json [hunk 1]
```diff
@@ -471,7 +471,8 @@
       },
       "openScenario": "Le scénario n'a pu etre ouvert.",
       "noPermissionDetail": "Vous n'avez pas la permission d'accéder au scénario #{{scenarioId}} du workspace #{{workspaceId}} de l'organisation #{{organizationId}}",
-      "datasetNotDeleted": "Le dataset n'a pas pu être supprimé"
+      "datasetNotDeleted": "Le dataset n'a pas pu être supprimé",
+      "datasetNotCreated": "Le dataset n'a pas pu être crée"
```
Comment (csm-thu) [inline public/locales/fr/translation.json:475]: Typo: `crée` -> `créé`

File: src/state/sagas/datasets/CreateDataset/CreateDataset.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
+import { t } from 'i18next';
+
+const getDatasets = (state) => state.dataset.list.data;
+
+export function* createDataset(action) {
+  const dataset = action.dataset;
+  const organizationId = action.organizationId;
+  try {
+    const datasets = [...(yield select(getDatasets))];
```
Comment (csm-thu) [inline src/state/sagas/datasets/CreateDataset/CreateDataset.js:16]: Copying the whole list of datasets may lead to performance issues when we have 15k+ datasets. Maybe we could use the action `ADD_DATASET` to simply add the new dataset to the list?

File: public/locales/fr/translation.json (hunk 1 above)
Comment (esasova) [reply public/locales/fr/translation.json:475]: :woman_facepalming:

File: src/state/sagas/datasets/CreateDataset/CreateDataset.js (hunk 2 above)
Comment (esasova) [reply src/state/sagas/datasets/CreateDataset/CreateDataset.js:16]: good idea, thanks ! :pray:

**Review by csm-thu — APPROVED** [review]
LGTM :+1:

**Review by nborde-CSM — APPROVED** [review]
LGTM :+1:

**Pull request merged**